internal/host/host_test.go

```diff
@@ -945,16 +970,80 @@ var _ = Describe("UpdateInventory", func() {
 		common.DeleteTestDB(db, dbName)
 	})
 
+	It("Test populate eligibility", func() {
+		testDisks := make([]*models.Disk, 0)
+		expectedDisks := make([]*models.Disk, 0)
+
+		for _, test := range []struct {
+			agentDecision    bool
+			agentReasons     []string
+			serviceDecision  bool
+			serviceReasons   []string
+			expectedDecision bool
+			expectedReasons  []string
+		}{
+			{false, []string{}, true, []string{}, true, []string{}},
+			{false, []string{}, false, []string{"Service reason"}, false, []string{"Service reason"}},
+			{true, []string{}, true, []string{}, true, []string{}},
+			{true, []string{}, false, []string{"Service reason"}, false, []string{"Service reason"}},
+			{false, []string{"Agent reason"}, true, []string{}, false, []string{"Agent reason"}},
+			{false, []string{"Agent reason"}, false, []string{"Service reason"}, false, []string{"Agent reason", "Service reason"}},
+			{true, []string{"Agent reason"}, true, []string{}, true, []string{}},
+			{true, []string{"Agent reason"}, false, []string{"Service reason"}, false, []string{"Service reason"}},
+		} {
+			testDisk := &models.Disk{
+				Eligibility: &models.DiskEligibility{Eligible: test.agentDecision, NotEligibleReasons: test.agentReasons},
+			}
+
+			testDisks = append(testDisks, testDisk)
+
+			mockValidator.EXPECT().DiskIsEligible(gomock.Any()).Return(test.serviceDecision, test.serviceReasons)
+
+			expectedDisks = append(expectedDisks,
+				&models.Disk{
+					Eligibility: &models.DiskEligibility{Eligible: test.expectedDecision, NotEligibleReasons: test.expectedReasons},
+				},
+			)
+		}
+
+		testInventory, err := json.Marshal(&models.Inventory{Disks: testDisks})
+		Expect(err).To(BeNil())
+
+		var actualInventory models.Inventory
+		populated, err := hapi.(*Manager).populateDisksEligibility(string(testInventory))
+		Expect(err).To(BeNil())
+		err = json.Unmarshal([]byte(populated), &actualInventory)
+		Expect(err).To(BeNil())
+
+		Expect(actualInventory.Disks).Should(Equal(expectedDisks))
+	})
+
 	Context("enable host", func() {
-		newInventory := "new inventory stuff"
-		success := func(reply error) {
-			Expect(reply).To(BeNil())
+		var newInventoryBytes []byte
+
+		BeforeEach(func() {
+			// Create an inventory that is slightly arbitrarily different than the default one
```

Comment:
The reason this modification was required is because now the UpdateInventory (the function being tested) actually parses and modifies the JSON, rather than treat is as a blob. So the test `"new inventory stuff"` blob was causing problems because it was not valid JSON.